src/coreclr/jit/inductionvariableopts.cpp

```diff
@@ -751,6 +751,77 @@ void Compiler::optBestEffortReplaceNarrowIVUses(
     });
 }
 
+//------------------------------------------------------------------------
+// optWidenIVs: Widen IVs of the specified loop.
+//
+// Parameters:
+//   scevContext - Context for scalar evolution
+//   loop        - The loop
+//   loopLocals  - Data structure for locals occurrences
+//
+// Returns:
+//   True if any primary IV was widened.
+//
+bool Compiler::optWidenIVs(ScalarEvolutionContext& scevContext,
+                           FlowGraphNaturalLoop*   loop,
+                           LoopLocalOccurrences*   loopLocals)
+{
```

Comment:
This is just factoring IV widening into its own function... the new pass is down below.